Source/JavaScriptCore/dfg/DFGSpeculativeJIT64.cpp

```diff
@@ -6823,6 +6824,121 @@ void SpeculativeJIT::compileEnumeratorPutByVal(Node* node)
     noResult(node);
 }
 
+void SpeculativeJIT::compileCreateClonedArguments(Node* node)
+{
+    if (m_graph.isWatchingHavingABadTimeWatchpoint(node)) {
+        JSGlobalObject* globalObject = m_graph.globalObjectFor(node->origin.semantic);
+
+        GPRTemporary storage(this);
+        GPRTemporary result(this);
+        GPRTemporary size(this);
+        GPRTemporary scratch(this);
+        GPRTemporary scratch2(this);
+
+        GPRReg storageGPR = storage.gpr();
+        GPRReg resultGPR = result.gpr();
+        GPRReg sizeGPR = size.gpr();
+        GPRReg scratchGPR = scratch.gpr();
+        GPRReg scratch2GPR = scratch2.gpr();
+
+        JumpList slowCases;
+
+        emitGetLength(node->origin.semantic, sizeGPR);
+
+        slowCases.append(branch32(AboveOrEqual, sizeGPR, TrustedImm32(MAX_STORAGE_VECTOR_LENGTH)));
+
+        {
+            // ClonedArguments' butterfly requires property storage for "length".
+            size_t outOfLineCapacity = globalObject->clonedArgumentsStructure()->outOfLineCapacity();
+            static_assert((1U << 3) == sizeof(JSValue));
+            lshift32(sizeGPR, TrustedImm32(3), scratchGPR);
+            add32(TrustedImm32(sizeof(IndexingHeader) + outOfLineCapacity * sizeof(JSValue)), scratchGPR, scratch2GPR);
+            move(TrustedImmPtr(nullptr), storageGPR);
```

Comment:
Do we really need to set `storageGPR` to nullptr here given that `emitAllocateVariableSized` is going to set it immediately below?

Oh, I see ... the `slowCases` path relies on `storageGPR` being initialized.  However, you have a branch above for the `sizeGPR` vs `MAX_STORAGE_VECTOR_LENGTH` comparison.  I suppose it doesn't matter because the slow path operation will do a length check first.  However, it does feel a little brittle to pass an uninitialized pointer like this.  Can we move this initialization of `storageGPR` above the branch above?

Reply:
Moved.